keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -117,59 +116,44 @@ serves the resourceVersion="0" list requests from reflectors today.
 
 Guard this by a `WatchCacheConsistentReads` feature gate.
 
-#### Use WithProgressNotify to enable automatic watch updates
+This alternative requires using `WatchProgressRequest` which is only available in etcd 3.4+, and so would
+require we make the kube-apiserver aware of etcd's minor version, which is described in more detail later.
 
-Create etcd watches with `WithProgressNotify` enabled (available in all etcd 3.x versions).
+<<[UNRESOLVED @serathius]>>
+Propose a method to make kube-apiserver aware of etcd version:
+* Flag providing etcd version to kube-apiserver `--storage-backend=etcd3.4`
+* Flag to enable/disable the feature `--allow-using-progress-notify`
+* Have kube-apiserver check cluster version in etcd `/version` endpoint. 
+  Retry the check logic if `WatchProgressRequest` fails.
+<<[/UNRESOLVED]>>
```

Comment:
You mentioned to me offline that the decision during SIG api-machinery meeting was basically to fallback to regular Range request to etcd if the Progress request from watcher fails.

But I just realized that https://github.com/etcd-io/etcd/issues/15220 is making this problematic - there are versions that support those requests, but those doesn't provide us enough guarantees.
@deads2k @jpbetz - thoughts?

Reply:
https://github.com/etcd-io/etcd/issues/15220 causes a race between sending an event and progress notification with the same revision. It can create a situation where client gets watch notification for revision X, before it gets event with revision X. This allows for situation where watch will break on watch notification and client restoring watch on revision X+1 will skip the event, however I think the chances of that are minimal as those two watch responses are send immediately one after the other.

Still, there is a risk. However, I don't think we should block development on it. For Alpha and Beta we can note this in documentation of the Feature flag to ensure users are aware of the risk. Any suggestions for GA?